master/docs/manual/cfg-schedulers.rst

```diff
@@ -35,6 +35,83 @@ There are several common arguments for schedulers, although not all are availabl
 
 ``builderNames``
     This is the set of builders which this scheduler should trigger, specified as a list of names (strings).
+    This can also be an :class:`~IRenderable` object which will render to a list of builder names (or a list of :class:`~IRenderable` that will render to builder names).
+
+    .. note:: When ``builderNames`` is rendered, these additional :class:`~Properties` attributes are available:
+
+       ``master``
+           A reference to the :class:`~BuildMaster` object that owns this scheduler.
+           This can be used to access the data API.
+       ``sourcestamps``
+           The list of sourcestamps that triggered the scheduler.
+       ``changes``
+           The list of changes associated with the sourcestamps.
+       ``files``
+           The list of modified files associated with the changes.
+
+       Any property attached to the change(s) that triggered the scheduler will be combined and available when rendering `builderNames`.
+
+    Here is a simple example:
+
+    .. code-block:: python
+
+       from buildbot.plugins import util, schedulers
+
+       @util.renderer
+       def builderNames(props):
+           builders = []
+           for f in props.files:
+               if f.endswith('.rst'):
+                   builders.append('check_docs')
```

Comment:
Actually, this code is shitty (I know I wrote it). We should use a set in order to deduplicate (and turn into a list in return statement)